contracts/src/token/erc721/extensions/uri_storage.rs

```diff
@@ -85,14 +85,14 @@ mod tests {
     }
 
     #[motsu::test]
-    fn set_token_uri_works(contract: Erc721UriStorage) {
+    fn set_token_uri_works(contract: Erc721URIStorage) {
         let token_id = random_token_id();
 
         let initial_token_uri = String::from("https://docs.openzeppelin.com/contracts/5.x/api/token/erc721#Erc721URIStorage");
         contract._token_uris.setter(token_id).set_str(initial_token_uri);
 
         let token_uri = String::from("Updated Token URI");
```

Comment:
Could you expand more on this @alexfertel ?

Reply:
`String` is the owned, allocated variant of `str`. This means that `String` lives on the heap and goes through an alloc/free lifecycle. `str` on the other hand is just a slice of bytes: where it lives is an implementation detail, but usually it's put in read-only memory or in the stack. `&str` always lives in the stack. It doesn't allocate. 

`String` and `&str` are convertible between each other, however `String` -> `&str` is almost free, because it's, roughly, borrowing the internal `String` buffer. Think of it as `&[u8]` and `Vec<u8>` or `&Path` and `PathBuf`. Most of the time, using the borrowed variants will be more performant and generate fewer instructions.

This is one of the reasons we always accept the borrowed variants as function parameters:

- If the function takes a `&str` and it receives a `&str`, all good.
- If the function takes a `&str` and it receives a `String`, easy conversion, just borrow the internal buffer. We know this is sound, since the caller will go out of scope after the callee.
- If the function takes a `String` it can't receive a `&str`: we are expecting the owned version.

This is also why `&String` doesn't make sense, and signals a misuse. You have an unnecessary layer of indirection: `&String` -> `String` -> `&str`. Whenever you have a `&String` you can equivalently have a `&str` and it will be more performant and flexible.

Reply:
I know the difference between `String`, `&str` and `&String` and leaving `&String` was my oversight. 
My question was should I change sth in the API of the  `UriStorage` extension or you meant only the change in E2E tests from this PR.

Reply:
> My question was should I change sth in the API of the UriStorage extension or you meant only the change in E2E tests from this PR.

No, that looks great! I was referring to the strings.